modules/recording/recording/libs/audio-recorder.ts

```diff
@@ -1,66 +1,76 @@
 export class AudioRecorder {
   private mediaRecorder: MediaRecorder | null = null;
   private audioChunks: Blob[] = [];
-  private stream: MediaStream | null = null;
-
-  async startRecording(deviceId?: string): Promise<void> {
-    if (this.mediaRecorder && this.mediaRecorder.state === "paused") {
-      this.mediaRecorder.resume();
-      return;
-    }
+  private audioContext: AudioContext | null = null;
+  private sourceNode: MediaStreamAudioSourceNode | null = null;
 
+  async startRecording(deviceId: string) {
     try {
-      const constraints: MediaStreamConstraints = {
-        audio: deviceId ? { deviceId: { exact: deviceId } } : true,
-      };
-      this.stream = await navigator.mediaDevices.getUserMedia(constraints);
-      this.mediaRecorder = new MediaRecorder(this.stream);
+      const stream = await navigator.mediaDevices.getUserMedia({
+        audio: { deviceId: deviceId ? { exact: deviceId } : undefined },
+      });
+
+      this.audioContext = new (window.AudioContext ||
+        (window as any).webkitAudioContext)();
+      this.sourceNode = this.audioContext.createMediaStreamSource(stream);
+
+      this.mediaRecorder = new MediaRecorder(stream);
       this.audioChunks = [];
+
       this.mediaRecorder.ondataavailable = (event) => {
-        if (event.data.size > 0) {
-          this.audioChunks.push(event.data);
-        }
+        this.audioChunks.push(event.data);
       };
+
       this.mediaRecorder.start();
     } catch (error) {
       console.error("Error starting recording:", error);
-      throw new Error(`Failed to start recording: ${(error as Error).message}`);
+      throw error;
```

Comment:
Dont use `throw` inside the catch. 

Do we have a catch on the parent? Or catch it here or outside not in both